### PR TITLE
Handle empty value for config set proxy

### PR DESCRIPTION
**Pull request by dheyay**
## Why is this needed?

Currently the user can set config proxy value to empty using `pro config set http_proxy=`
This fixes that, since the given proxy is invalid, an error is shown.

Fixes: [#2925]

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [ ] No

### CODE REVIEW

**Comment by dheyay** [comment]
Might need updated translations, also open to suggestions for a solution to re-use other error messages than creating a new one

File: uaclient/messages/__init__.py [hunk 1]
```diff
@@ -2425,6 +2425,11 @@ def __repr__(self):
     "invalid-arg-choice", "\n" + t.gettext("{arg} must be one of: {choices}")
 )
 
+E_CLI_VALID_VALUE = FormattedNamedMessage(
```
Comment (orndorffgrant) [inline uaclient/messages/__init__.py:2428]:
```suggestion
E_CLI_EMPTY_CONFIG_VALUE = FormattedNamedMessage(
```

File: uaclient/exceptions.py [hunk 2]
```diff
@@ -496,6 +495,10 @@ class InvalidArgChoice(UbuntuProError):
     _formatted_msg = messages.E_CLI_VALID_CHOICES
 
 
+class InvalidArgValue(UbuntuProError):
+    _formatted_msg = messages.E_CLI_VALID_VALUE
```
Comment (orndorffgrant) [inline uaclient/exceptions.py:499]:
```suggestion
class EmptyConfigValue(UbuntuProError):
    _formatted_msg = messages.E_CLI_EMPTY_CONFIG_VALUE
```

File: uaclient/cli/__init__.py [hunk 3]
```diff
@@ -690,6 +690,9 @@ def action_config_set(args, *, cfg, **kwargs):
         raise exceptions.InvalidArgChoice(
             arg="<key>", choices=", ".join(config.UA_CONFIGURABLE_KEYS)
         )
+    if not set_value.strip():
+        subparser.print_help()
+        raise exceptions.InvalidArgValue(arg=set_key)
```
Comment (orndorffgrant) [inline uaclient/cli/__init__.py:695]:
```suggestion
        raise exceptions.EmptyConfigValue(arg=set_key)
```

**Review by orndorffgrant — COMMENTED** [review]
Thanks!
Let's name the error and message more specifically.
Otherwise, looks good!

**Comment by dheyay** [comment]
Updated @orndorffgrant

**Comment by renanrodrigo** [comment]
I am sad about this nit, but:
could you just adapt the commit message comment to say
`Fixes: #2925`
instead of 
`Fixes Bug: [#2925]`

For CI reasons, we always use that format :/

File: uaclient/cli/__init__.py [hunk 4]
```diff
@@ -690,6 +690,9 @@ def action_config_set(args, *, cfg, **kwargs):
         raise exceptions.InvalidArgChoice(
             arg="<key>", choices=", ".join(config.UA_CONFIGURABLE_KEYS)
         )
+    if not set_value.strip():
+        subparser.print_help()
```
Comment (lucasmoura) [inline uaclient/cli/__init__.py:694]: Do we need to print the help message here ? Maybe just stating that the user provided an invalid empty value is enough.

What do you think @renanrodrigo ?

Comment (renanrodrigo) [reply uaclient/cli/__init__.py:694]: This is following what we do for all other errors on setting config values. I think this looks good as is.

**Review by renanrodrigo — APPROVED** [review]
LGTM

**Pull request merged**